eth_keyfile/keyfile.py

```diff
@@ -229,7 +273,116 @@ def _decode_keyfile_json_v3(keyfile_json: Dict[str, Any], password: bytes) -> by
 
 
 #
-# Verson 4 decoder
+# Version 4 creator
+#
+
+
+def _create_v4_keyfile_json(
+    private_key: Union[bytes, bytearray, memoryview],
+    password: bytes,
+    kdf: KDFType,
+    work_factor: Optional[int] = None,
+    salt_size: Optional[int] = None,
+    description: Optional[str] = None,
+    path: Optional[str] = None,
+) -> Dict[str, Any]:
+    if work_factor is None:
+        work_factor = get_default_work_factor_for_kdf(kdf)
+    if salt_size is None:
+        salt_size = 32
+    if description is None:
+        description = ""
+    if path is None:
+        path = ""
+
+    aes_iv = Random.get_random_bytes(16)
+
+    if to_int(private_key) > MAX_V4_PRIVATE_KEY:
+        raise EthKeyfileValueError("Invalid `private_key`, exceeds maximum valid value")
```

Comment:
Nit: Could also add the max value in this message

Reply:
done!